src/sage/matroids/basis_exchange_matroid.pyx

```diff
@@ -1292,19 +1255,17 @@ cdef class BasisExchangeMatroid(Matroid):
                 self._whitney_numbers2_rec(f_vec, flats, todo, e + 1, i + 1)
             e = bitset_next(todo[i], e)
 
-    cpdef flats(self, r):
+    cpdef SetSystem flats(self, long k):
```

Comment:
Technically this is a breaking change because user code may be passing `r` as a keyword argument

Reply:
What to do for this? `@rename_keyword` does not work for a `cpdef` function I think.

Reply:
It would be OK with me to just argue that given the very short name of this parameter, it was intended to be a positional-only argument but that Cython has historically not supported declaring positional-only arguments (do you know if it supports them now?), and therefore renaming without deprecation is fine.

But I'd welcome a second opinion on this; @tscrim?

Reply:
This argument seems convincing to me too. I don't know about the support of positional-only arguments.

Reply:
I highly doubt that anyone is using it as a keyword argument, but we should properly deprecate it if changing the name. Of course, we don't strictly have to do this, but it is the general policy. My preference would be to just keep the current argument name, but I think it is okay to go ahead and do it (without a deprecation) if you feel the renaming is warranted.

Reply:
I would like to rename it please. :D

In fact I would like to eventually incorporate the functionality of `independent_k_sets()` into `independent_sets()` with an optional argument `k`. Also I want to make `flats()` with no argument return all flats.

Let me know if I should do these here.

Reply:
Some justification would be good all the same. Right now it seems the name change is quite arbitrary (with `r` perhaps being sort for `rank`).

One reason for having separate methods is that the logic is more simple (and so marginally faster, which could matter in a tight loop). However, I would side with you on having just one method makes the API easier and that extra check/argument parsing likely is inconsequential.

Reply:
`k` seems good to me for cardinalities (`k`-sets). `r` can be confused with the matroid's rank. Also, `k` is often not in use such as other indices (e.g., `i`).

Your concern about potential inefficiency is legitimate, however in this case it is indeed inconsequential as there is a lot of computation within the `if` statement.

Reply:
I'll leave the change of the functionality of flats for another PR.